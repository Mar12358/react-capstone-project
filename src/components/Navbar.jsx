import { NavLink, useLocation } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Icon from '../Assets/back_icon.png';
import styles from '../styles/Navbar.module.css';
import { filter } from '../redux/assets/assetsSlice';

const link = { path: '/assets', text: 'Assets' };

const Navbar = () => {
  const dispatch = useDispatch();
  const location = useLocation();
  const currentPath = location.pathname.split('/').pop();
  const capitalizedPath = currentPath.charAt(0).toUpperCase() + currentPath.slice(1);

  return (
    <nav className={styles.wrapper}>
      <div className={styles.navbar}>
        <div className={styles.navHeader}>
          <div className={styles.arrowContainer}>
            <NavLink
              to={link.path}
              className={currentPath === 'assets' ? styles.hidden : styles.visible}
            >
              <img className={styles.logoIcon} src={Icon} alt="Icon" />
            </NavLink>
          </div>
          <div className={styles.hContainer}>
            <h1>CoinCap</h1>
            <h2>{capitalizedPath}</h2>
          </div>
          <div />
        </div>
        <div className={styles.inputContainer}>
          <input
            type="text"
            className={styles.input}
            placeholder="Search by Asset name..."
            onChange={(e) => dispatch(filter(e.target.value))}
          />
        </div>
      </div>
      <div className={styles.separator} />
    </nav>
  );
};
export default Navbar;
