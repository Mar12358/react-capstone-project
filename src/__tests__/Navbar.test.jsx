import configureStore from 'redux-mock-store';
import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import assetsReducer, { filter } from '../redux/assets/assetsSlice';

const mockStore = configureStore([thunk]);
jest.mock('axios');

describe('Navbar.jsx renders propperly', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Navbar />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Search Bar filters correctly (Reducer Works)', () => {
  it('should handle the filter action', () => {
    const initialState = {
      assets: {
        data: [{ name: 'bitcoin', id: 'bitcoin' },
          { name: 'ethereum', id: 'ethereum' }],
      },
    };

    const action = filter('bitcoin');
    const newState = assetsReducer(initialState, action);
    expect(newState.filterData.some((obj) => obj.name === 'bitcoin')).toBe(true);
    expect(newState.filterData.some((obj) => obj.name === 'ethereum')).toBe(false);
  });
});
