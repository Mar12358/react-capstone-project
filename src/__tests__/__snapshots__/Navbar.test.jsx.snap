// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.jsx renders propperly renders correctly 1`] = `
<nav
  className="wrapper"
>
  <div
    className="navbar"
  >
    <div
      className="navHeader"
    >
      <div
        className="arrowContainer"
      >
        <a
          className="visible"
          href="/assets"
          onClick={[Function]}
        >
          <img
            alt="Icon"
            className="logoIcon"
            src="back_icon.png"
          />
        </a>
      </div>
      <div
        className="hContainer"
      >
        <h1>
          CoinCap
        </h1>
        <h2 />
      </div>
      <div />
    </div>
    <div
      className="inputContainer"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search by Asset name..."
        type="text"
      />
    </div>
  </div>
  <div
    className="separator"
  />
</nav>
`;
